src/lib/dialog/dialog-container.ts

```diff
@@ -77,7 +80,8 @@ export class MdDialogContainer extends BasePortalHost implements OnDestroy {
     private _ngZone: NgZone,
     private _renderer: Renderer,
     private _elementRef: ElementRef,
-    private _focusTrapFactory: FocusTrapFactory) {
+    private _focusTrapFactory: FocusTrapFactory,
+    @Optional() @Inject(DOCUMENT) private _document: any) {
```

Comment:
We can add a type for this while working around Angular Universal's limitations by doing `Document|Document`